Bottom: 3f518c50d8354809224e4ab4b34e2a670885a71a
Top:    b7cd7f5f9d85f1b197f11db2a4462ee3ca681bfe
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-12-18 17:24:55 +0000

Code fixes


---

diff --git a/src/hid/common/hidnogui.c b/src/hid/common/hidnogui.c
index 9d88086..b08d90d 100644
--- a/src/hid/common/hidnogui.c
+++ b/src/hid/common/hidnogui.c
@@ -381,7 +381,7 @@ nogui_prompt_for (const char *msg, const char *default_string)
   if (answer == NULL)
     return strdup ((default_string != NULL) ? default_string : "");
   else
-    return strdup (answer);
+    return answer;
 }
 
 /* FIXME - this could use some enhancement to actually use the other
@@ -402,7 +402,7 @@ nogui_fileselect (const char *title, const char *descr,
   if (answer == NULL)
     return (default_file != NULL) ? strdup (default_file) : NULL;
   else
-    return strdup (answer);
+    return answer;
 }
 
 static int
diff --git a/src/hid/gtk/ghid-layer-selector.c b/src/hid/gtk/ghid-layer-selector.c
index 432e57e..2b04063 100644
--- a/src/hid/gtk/ghid-layer-selector.c
+++ b/src/hid/gtk/ghid-layer-selector.c
@@ -643,7 +643,7 @@ ghid_layer_selector_add_layer (GHidLayerSelector *ls,
       free_ldata (ls, new_layer);
     }
 
-  new_layer = malloc (sizeof (*new_layer));
+  new_layer = g_malloc (sizeof (*new_layer));
 
   gtk_list_store_set (ls->list_store,
                       &iter,
